src/data/list/defs.lean

```diff
@@ -166,6 +166,14 @@ def find (p : α → Prop) [decidable_pred p] : list α → option α
 | []     := none
 | (a::l) := if p a then some a else find l
 
+/-def list.mfirst {m : Type u → Type v} [monad m] [alternative m] {α : Type w} {β : Type u} (f : α → m β) : list α → m β
+| []      := failure
+| (a::as) := f a <|> list.mfirst as
+-/
+/-- `mfind tac l` returns the first element of `l` on which `tac` succeeds, and fails otherwise. -/
+meta def mfind {α} {m : Type → Type} [monad m] [alternative m] (tac : α → m unit) : list α → m α :=
```

Comment:
Why is this meta and not universe polymorphic?

Reply:
`meta` because I generalized it from the `tactic` version and forgot to remove the keyword. It can only be universe polymorphic in the output type because of the `m unit`; I'll change that.